tools/speech_data_explorer/data_explorer.py

```diff
@@ -1,846 +1,1044 @@
+                selected_rows=[0],
+                page_action='custom',
+                page_current=0,
+                page_size=DATA_PAGE_SIZE,
+                page_count=math.ceil(len(data) / DATA_PAGE_SIZE),
+                style_cell={'overflow': 'hidden', 'textOverflow': 'ellipsis', 'maxWidth': 0, 'textAlign': 'center'},
+                style_header={
+                    'color': 'text-primary',
+                    'text_align': 'center',
+                    'height': 'auto',
+                    'whiteSpace': 'normal',
+                },
+                css=[
+                    {'selector': '.dash-spreadsheet-menu', 'rule': 'position:absolute; bottom: 8px'},
+                    {'selector': '.dash-filter--case', 'rule': 'display: none'},
+                    {'selector': '.column-header--hide', 'rule': 'display: none'},
+                ],
+            ),
+        )
+    ),
+] + [
+    dbc.Row(
+        [
+            dbc.Col(
+                html.Div(children=k.replace('_', ' ')),
+                width=2,
+                class_name='mt-1 bg-light font-monospace text-break small rounded border',
+            ),
+            dbc.Col(html.Div(id='_' + k), class_name='mt-1 bg-light font-monospace text-break small rounded border'),
+        ]
+    )
+    for k in data[0]
+]
+
+if metrics_available:
+    samples_layout += [
+        dbc.Row(
+            [
+                dbc.Col(
+                    html.Div(children='text diff'),
+                    width=2,
+                    class_name='mt-1 bg-light font-monospace text-break small rounded border',
+                ),
+                dbc.Col(
+                    html.Iframe(
+                        id='_diff',
+                        sandbox='',
+                        srcDoc='',
+                        style={'border': 'none', 'width': '100%', 'height': '100%'},
+                        className='bg-light font-monospace text-break small',
+                    ),
+                    class_name='mt-1 bg-light font-monospace text-break small rounded border',
+                ),
+            ]
+        )
+    ]
+samples_layout += [
+    dbc.Row(dbc.Col(html.Audio(id='player', controls=True),), class_name='mt-3 '),
+    dbc.Row(dbc.Col(dcc.Graph(id='signal-graph')), class_name='mt-3'),
+]
+
+
+@app.callback(
+    [Output('datatable', 'data'), Output('datatable', 'page_count')],
+    [Input('datatable', 'page_current'), Input('datatable', 'sort_by'), Input('datatable', 'filter_query')],
+)
+def update_datatable(page_current, sort_by, filter_query):
+    data_view = data
+    filtering_expressions = filter_query.split(' && ')
+    for filter_part in filtering_expressions:
+        col_name, op, filter_value = split_filter_part(filter_part)
+
+        if op in ('eq', 'ne', 'lt', 'le', 'gt', 'ge'):
+            data_view = [x for x in data_view if getattr(operator, op)(x[col_name], filter_value)]
+        elif op == 'condtains':

```

Comment:
Please fix the introduced typo: `condtains` => `contains`

Reply:
done